explorer/testdata/for/for-loop-auto.carbon

```diff
@@ -0,0 +1,25 @@
+// Part of the Carbon Language project, under the Apache License v2.0 with LLVM
```

Comment:
For consistency with the other `.carbon` files, I think these files' names should use `_` instead of `-`.